### DISCUSSION TITLE
automerging blocked by nonexistent checks

### DISCUSSION THREAD

**Discussion by evanstoddard23** [renovatebot/renovate#8891] [Request Help]
I'm using the GitHub hosted Renovate app. I'm noticing some PRs that pass all checks and have automerge enabled still (even days later) don't get automerged. Looking at the job logs I see:
```DEBUG: branch status check result(branch="renovate/commitizen-tools-commitizen-2.15.x")
{
  "state": "pending",
  "statuses": []
}
DEBUG: check runs result(branch="renovate/commitizen-tools-commitizen-2.15.x")
{
  "checkRuns": [
    {
      "name": "pre-commit",
      "status": "completed",
      "conclusion": "success"
    },
    {
      "name": "commitlint",
      "status": "completed",
      "conclusion": "success"
    },
    {
      "name": "tests",
      "status": "completed",
      "conclusion": "failure"
    },
    {
      "name": "pre-commit",
      "status": "completed",
      "conclusion": "success"
    }
  ]
}
DEBUG: PR is not ready for merge (branch status is red)(branch="renovate/commitizen-tools-commitizen-2.15.x")
```

These are all GitHub Actions configured checks I have set up in this repo, but "tests" is only triggered to run when select files are modified as it is a costlier operation. The GitHub UI looks totally clean and does not show this "tests" check in the PR (as the selected files are not modified). Is there any way I can still get Renovate to automerge this PR without running the "tests" check?

<img width="476" alt="Screen Shot 2021-02-26 at 4 21 10 PM" src="https://user-images.githubusercontent.com/2703526/109362124-01550d00-7850-11eb-8bf2-373f00be2fbb.png">

**Comment by rarkins** [comment]
No, it's not possible if a check is reporting "failed" but it would be possible if it was conclusion "skipped". Can you reproduce the problem in a simple public repo?

**Reply by evanstoddard23** [reply]
@rarkins thanks for the suggestion. I have modified my workflows to always run but skip certain jobs. Now in the job log I don't see the check failure being reported anymore, but it's still not automerging the PR. The only thing I can think that might cause this is there is an additional non-renovate commit on the PR, but I don't see any mention in the docs of this being an issue as long as all the status checks are passing/skipping. I don't have an easy way to set this up publicly but can if we reach a dead end.

**Reply by evanstoddard23** [reply]
Looking further at the history of automerged PRs seems to support that only those without non-renovate-created commits will be automerged. Is this the case? I feel I've looked at most of the [docs around automerging](https://docs.renovatebot.com/automerge-configuration/) but don't see this mentioned anywhere.

**Reply by rarkins** [reply]
We need to update the docs then. We definitely intentionally skip automerging once additional git authors are found, because a PR with passing checks doesn't meant it's necessarily ready/finished. The PR itself should contain a mention in its description that rebasing is stopped due to the PR being modified, but I'm not sure if it's mentioned in the rebase description. 

There's a recent feature request for "ignoreAuthors" which may be applicable here.

**Reply by evanstoddard23** [reply]
There is a message in the PR about it not automatically rebasing with the additional commits, but I think the connection to automerging there could use more clarity in the docs. Do you have a link by chance to that feature request? Thanks for your help!

**Reply by viceice** [reply]
simple search shows this #8893
